pkg/security/module/cws.go

```diff
@@ -194,22 +196,28 @@ func (c *CWSConsumer) Start() error {
 
 	// we can now wait for self test events
 	cb := func(success []eval.RuleID, fails []eval.RuleID, testEvents map[eval.RuleID]*serializers.EventSerializer) {
-		seclog.Debugf("self-test results : success : %v, failed : %v, retry %d/%d", success, fails, c.selfTestRetry.Load()+1, maxSelftestRetry)
+		c.selfTestCount++
 
-		if len(fails) > 0 && c.selfTestRetry.Load() < maxSelftestRetry {
-			c.selfTestRetry.Inc()
+		seclog.Debugf("self-test results : success : %v, failed : %v, run %d", success, fails, c.selfTestCount)
 
-			time.Sleep(selftestDelay)
-
-			if _, err := c.RunSelfTest(false); err != nil {
-				seclog.Errorf("self-test error: %s", err)
-			}
-			return
+		delay := selftestDelay
+		if c.selfTestPassed {
+			delay = selftestPassedDelay
 		}
 
-		if c.config.SelfTestSendReport {
+		if !c.selfTestPassed && c.selfTestCount == selftestMaxRetry {
+			c.reportSelfTest(success, fails, testEvents)
+		} else if len(fails) == 0 {
+			c.selfTestPassed = true
+
 			c.reportSelfTest(success, fails, testEvents)
 		}
```

Comment:
Shouldn't we keep honoring the `c.config.SelfTestSendReport` flag?

Reply:
Done